src/main/java/frc/robot/subsystems/DriveBase.java

```diff
@@ -32,7 +32,7 @@
     private DriveMode driveMode;
 
     // The drivebase talons
-    private WPI_TalonSRX leftFrontTalon, leftRearTalon, rightFrontTalon, rightRearTalon;
+    public WPI_TalonSRX leftFrontTalon, leftRearTalon, rightFrontTalon, rightRearTalon;
```

Comment:
Ah yes. Because he needed to access the talon object in order to pass it to the Pigeon constructor, since that is how you tell the Pigeon that it is attached to a Talon.

My question is - if the Pigeon and the Talon are so strongly interrelated -- AND the Pigeon is physically part of the drivebase -- should the Pigeon be a member of DriveBase?

Reply:
agreed, see my previous comments about rearchitecting where to put the pigeon. If it needs to know about the Talon (which it does) then maybe it should be managed here along with the Talon.